backend/main_page/views/main_page_view.py

```diff
@@ -2,19 +2,34 @@
 from rest_framework.request import Request
 from rest_framework.response import Response
 from rest_framework.views import APIView
+from main_page.views.sentiment_analysis import SentimentAnalysis
 from main_page.views.word_search_view import WordSearch
 
-
 class MainPageView(APIView):
     @staticmethod
     def get(request: Request) -> Response:
+        # Use WordSearch view to get initial data
         word_searched: Response = WordSearch.as_view()(request)
 
         if word_searched.status_code == 404:
             return word_searched
 
-        aggregated_data = {
-            **word_searched.data,
-        }
+        # Assuming you have a text to analyze
+        text_to_analyze = "hello this analyses sentiments"
+
+        # Create an instance of the SentimentAnalysis class
+        sentiment_analysis_instance = SentimentAnalysis()
+
+        # Call the process_sentiment_analysis method
+        sentiment_analysis_result = sentiment_analysis_instance.process_sentiment_analysis(text_to_analyze)
 
-        return Response(aggregated_data, status=status.HTTP_200_OK)
+        if sentiment_analysis_result:
+            # Add sentiment analysis to aggregated_data
+            aggregated_data = {
+                **word_searched.data,
+                "sentiment_analysis": sentiment_analysis_result,
+            }
+            return Response(aggregated_data, status=status.HTTP_200_OK)
+        else:
+            # Handle error in sentiment analysis
+            return Response({"error": "Error in sentiment analysis"}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
Unnecessary code. The `MainPageView` should only have a call to your view, such as:
```python
sentiment_analysis = SentimentAnalysis.as_view()(request)

aggregated_data = {
    **word_searched.data,
    **sentiment_analysis,  # the data is not necessary to be added like this if you don't want
}
```